packages/angular/src/generators/library/library.ts

```diff
@@ -105,7 +122,11 @@ export async function libraryGenerator(tree: Tree, schema: Schema) {
       tree,
       {},
       {
-        'ng-packagr': ngPackagrVersion,
+        'ng-packagr':
+          userInstalledAngularVersion.major !== major(coerce(angularVersion))
+            ? versions[`angularV${userInstalledAngularVersion.major}`]
+                .ngPackagrVersion
+            : ngPackagrVersion,
```

Comment:
@leosvelperez I'm not 100% sold on this, but if we implement this pattern and follow it, it should be allow us to support multiple versions without nested `if` or ternary operators

Reply:
I think the pattern is ok, but there's an issue with the condition. It's assuming that when the major is different, it will always be a major we have on our map. It could be a version lower than what we support or it could be a greater version in cases where a new Angular major comes out and we still haven't added support (edge case). In both scenarios, it won't be found on the map and you'll get an error when trying to access `ngPackagrVersion` on `undefined`.

In both cases, we can add the latest version we have in the `versions.ts` file. That would be the same behavior we have today if someone tries to generate a library using an Angular version that's not supported (either lower or greater).

Something like the below should probably be what we need:

```suggestion
        'ng-packagr':
          userInstalledAngularVersion.major < major(coerce(angularVersion))
            ? versions[`angularV${userInstalledAngularVersion.major}`]
                ?.ngPackagrVersion ?? ngPackagrVersion
            : ngPackagrVersion,
```